### PR TITLE
Add definitions for atom-mocha-test-runner.

### PR DESCRIPTION
**Pull request by GlenCFL**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
The [atom-mocha-test-runner](https://www.npmjs.com/package/atom-mocha-test-runner) NPM package. The repository for the project is [here](https://github.com/BinaryMuse/atom-mocha-test-runner).
[Atom documentation on custom test runners.](http://flight-manual.atom.io/hacking-atom/sections/writing-specs/#customizing-your-test-runner)
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

This provides type definitions for atom-mocha-test-runner, while also adding type definitions for custom test runners to Atom.

### CODE REVIEW

File: types/atom-mocha-test-runner/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[*.ts]
+indent_style = tab
```
Comment (plantain-00) [inline types/atom-mocha-test-runner/.editorconfig]: This file shouldn't be included.

File: types/atom-mocha-test-runner/tslint.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        // Custom rules.
+        "class-name": true,
```
Comment (plantain-00) [inline types/atom-mocha-test-runner/tslint.json]: There should be no other rules here, it should be only `"extends": "dtslint/dt.json"`, 
if you really want to disable some rule, you can disable them in code

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/atom-mocha-test-runner/tslint.json (hunk 2 above)
Comment (GlenCFL) [reply types/atom-mocha-test-runner/tslint.json]: Is this actually written anywhere? These are extra enables, with zero disables.

**Review by plantain-00 — APPROVED** [review]

**Review by smhxx — APPROVED** [review]
Looks solid to me. 👍

**Pull request merged**